### DISCUSSION TITLE
how to receive skywalking grpc exporter data?

### DISCUSSION THREAD

**Discussion by houbinglei** [apache/skywalking#9764] [Q&A]
i used this proto file:

/* 

service MetricExportService {
    rpc export (stream ExportMetricValue) returns (ExportResponse) {
    }
    rpc subscription (SubscriptionReq) returns (SubscriptionsResp) {
    }
}

message ExportMetricValue {
    string metricName = 1;
    string entityName = 2;
    string entityId = 3;
    ValueType type = 4;
    int64 timeBucket = 5;
    int64 longValue = 6;
    double doubleValue = 7;
    repeated int64 longValues = 8;
}

message SubscriptionsResp {
    repeated SubscriptionMetric metrics = 1;
}

message SubscriptionMetric {
    string metricName = 1;
    EventType eventType = 2;
}

enum ValueType {
    LONG = 0;
    DOUBLE = 1;
    MULTI_LONG = 2;
}

enum EventType {
    // The metrics aggregated in this bulk, not include the existing persistent data.
    INCREMENT = 0;
    // Final result of the metrics at this moment.
    TOTAL = 1;
}

message SubscriptionReq {

}

message ExportResponse {
}
*/

**and i want use golang to write a grpc server , i have changed this proto to .go file~; but i dont konw how to write a server , and then to receive data from skywalking.
i have set this:**
/*
exporter:
  grpc:
    targetHost: 127.0.0.1
    targetPort: 9870

*/

**Comment by wu-sheng** [comment]
I don't know why this should be discussed at SkyWalking channel.
